users/service.go

```diff
@@ -200,6 +208,31 @@ func (svc service) ListClients(ctx context.Context, token string, pm mgclients.P
 	return pg, nil
 }
 
+func (svc service) SearchUsers(ctx context.Context, token string, pm mgclients.Page) (mgclients.ClientsPage, error) {
+	userID, err := svc.Identify(ctx, token)
+	if err != nil {
+		return mgclients.ClientsPage{}, err
+	}
+	if err := svc.checkSuperAdmin(ctx, userID); err == nil {
+		pg, err := svc.clients.SearchClients(ctx, pm)
+		if err != nil {
+			return mgclients.ClientsPage{}, errors.Wrap(svcerr.ErrViewEntity, err)
+		}
+		return pg, err
+	}
```

Comment:
No need for super admin check here 
we can support only search by `Name` i.e `Username` or `ID` and we should not support search by `Identity`.